contrib/images/test/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+# 
```

Comment:
Why did the directory move ?

If `check` is installed, it should generate all the test images, and then compile and execute the test suite under `/test`

Reply:
This way, it will be easy for the container to copy the test source code.

> If `check` is installed, it should generate all the test images, and then compile and execute the test suite under `/test`

Do you mean we need to integrate this test into the top-level `CMakeLists.txt` using `CTest`? 

If we want to succeed with this function, we need to fix all configurations; otherwise, the check won't work.

The test only works for container configurations. For user-custom configurations, it will not work.

Reply:
After we ensure Docker tests run well, I will create a custom command in the top-level `CMakeLists.txt` to allow users to call the Docker test. This command will create the image, run the test, and print the results. This is what I am thinking.

Reply:
I meant to have a `cmake/Findcheck.cmake` file that defines a `HAVE_CHECK` and the top-level `CMakeLists.txt` adds the `test/CMakeLists.txt` in order to compile the test suite if it is.

We can have the `Dockerfile`s in the `test/` directory to start with

Reply:
OK, we can do this, it’s easy. But I meant that it’s meaningless.

Although users can compile the test suite, they can’t use it.

The test suite only works in the container.

Reply:
Yeah, but we have to start somewhere

Reply:
Okay

Reply:
For now we have to create an infrastructure such that you can continue with the `extra` configuration property and test it...